roles/network_plugin/calico/tasks/install.yml

```diff
@@ -174,6 +185,7 @@
       "spec": {
           "logSeverityScreen": "Info",
           "nodeToNodeMeshEnabled": {{ nodeToNodeMeshEnabled|default('true') }} ,
+          "serviceExternalIPs": {{ _service_external_ips }},
```

Comment:
According to https://docs.projectcalico.org/networking/advertise-service-ips this way seems good for me.
One question: Is it fine to specify empty list for calico if calico_advertise_service_external_ips is the default []?

Reply:
Yes, that's fine. An empty list is an acceptable value for the serviceExternalIPs key.

For example, applied non-empty list of serviceExternalIPs:
```
~ cat bgp-conf-0.yaml
kind: BGPConfigurationList
apiVersion: projectcalico.org/v3
items:
- apiVersion: projectcalico.org/v3
  kind: BGPConfiguration
  metadata:
    name: default
  spec:
    asNumber: 65207
    nodeToNodeMeshEnabled: false
    serviceExternalIPs:
    - cidr: 10.108.13.0/24
~ calicoctl.sh apply -f bgp-conf-0.yaml
Successfully applied 1 'BGPConfiguration' resource(s)
~ calicoctl.sh get bgpconfig default -o yaml
apiVersion: projectcalico.org/v3
kind: BGPConfiguration
metadata:
  creationTimestamp: "2020-11-17T04:02:42Z"
  name: default
  resourceVersion: "1598124"
  uid: c205703f-cbeb-4b1e-94a6-468178ca63e3
spec:
  asNumber: 65207
  nodeToNodeMeshEnabled: false
  serviceExternalIPs:
  - cidr: 10.108.13.0/24
```

An empty list is also OK. In this case, it just sets the default behavior for serviceExternalIPs:
```
~ cat bgp-conf-1.yaml
kind: BGPConfigurationList
apiVersion: projectcalico.org/v3
items:
- apiVersion: projectcalico.org/v3
  kind: BGPConfiguration
  metadata:
    name: default
  spec:
    asNumber: 65207
    nodeToNodeMeshEnabled: false
    serviceExternalIPs: []
~ calicoctl.sh apply -f bgp-conf-1.yaml
Successfully applied 1 'BGPConfiguration' resource(s)
~ calicoctl.sh get bgpconfig default -o yaml
apiVersion: projectcalico.org/v3
kind: BGPConfiguration
metadata:
  creationTimestamp: "2020-11-17T04:02:42Z"
  name: default
  resourceVersion: "1598897"
  uid: c205703f-cbeb-4b1e-94a6-468178ca63e3
spec:
  asNumber: 65207
  nodeToNodeMeshEnabled: false
```

Reply:
Thanks for your explanation. I got it.